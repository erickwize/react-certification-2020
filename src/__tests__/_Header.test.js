import React from 'react';
import { screen, cleanup, render, fireEvent } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import HeaderMenu from '../components/Header';

afterEach(() => {
  cleanup();
});

const doSearch = jest.fn();

it('Should render Header', () => {
  const header = render(<HeaderMenu />);
  const headerComponent = screen.getByTestId('yt-header');
  expect(headerComponent).toBeInTheDocument();
  expect(headerComponent).toContainHTML('Dark mode');
  expect(headerComponent).toContainHTML('input');
  expect(header).toMatchSnapshot();
});

it('Toggle Header Dark mode', () => {
  render(<HeaderMenu />);

  const switchInput = screen.getByLabelText('Dark mode', { type: 'checkbox' });
  expect(switchInput).toBeInTheDocument();
  fireEvent(
    switchInput,
    new MouseEvent('click', {
      checked: true,
    })
  );
  expect(switchInput.checked).toBe(true);
  fireEvent(
    switchInput,
    new MouseEvent('click', {
      checked: false,
    })
  );
  expect(switchInput.checked).toBe(false);
});

it('Search value change', () => {
  render(<HeaderMenu doSearch={doSearch} />);

  const SearchInput = screen.getByPlaceholderText('Search...');
  expect(SearchInput).toBeInTheDocument();
  SearchInput.focus();
  fireEvent.change(SearchInput, { target: { value: 'wizeline' } });
  expect(SearchInput.value).toBe('wizeline');

  fireEvent.keyDown(SearchInput, { key: 'Enter', code: 'Enter' });
  setTimeout(() => {
    expect(doSearch).toHaveBeenCalledWith('wizeline');
  }, 10);

  fireEvent.blur(SearchInput, { key: 'Enter', code: 'Enter' });
  setTimeout(() => {
    act(() => {
      expect(doSearch).toHaveBeenCalledWith('wizeline');
    });
  }, 10);
});
