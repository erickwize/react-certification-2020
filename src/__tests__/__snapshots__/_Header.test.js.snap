// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bdfBwQ hIHNKi"
        data-testid="yt-header"
      >
        <div
          class="sc-gsTCUz ywWHk"
        >
          <a
            href="/"
          >
            <button
              class="sc-dlfnbm cOuLkH"
            />
          </a>
          <div
            class="sc-hKgILt ljSDGR"
          >
            <input
              class="sc-eCssSg iBGkVj"
              id="name"
              placeholder="Search..."
              type="text"
              value="wizeline"
            />
            <span
              class="sc-jSgupP hcyBed"
            />
          </div>
        </div>
        <div
          class="sc-gsTCUz ywWHk"
        >
          <div
            class="sc-gKsewC kjfNIn"
          >
            <input
              class="sc-pFZIQ gGzfcK"
              id="darkMode"
              name="darkMode"
              type="checkbox"
            />
            <label
              class="sc-iBPRYJ cpTqSp"
              for="darkMode"
            >
              Dark mode
            </label>
          </div>
          <button
            class="sc-fubCfw iHeQum"
          />
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="sc-bdfBwQ hIHNKi"
      data-testid="yt-header"
    >
      <div
        class="sc-gsTCUz ywWHk"
      >
        <a
          href="/"
        >
          <button
            class="sc-dlfnbm cOuLkH"
          />
        </a>
        <div
          class="sc-hKgILt ljSDGR"
        >
          <input
            class="sc-eCssSg iBGkVj"
            id="name"
            placeholder="Search..."
            type="text"
            value="wizeline"
          />
          <span
            class="sc-jSgupP hcyBed"
          />
        </div>
      </div>
      <div
        class="sc-gsTCUz ywWHk"
      >
        <div
          class="sc-gKsewC kjfNIn"
        >
          <input
            class="sc-pFZIQ gGzfcK"
            id="darkMode"
            name="darkMode"
            type="checkbox"
          />
          <label
            class="sc-iBPRYJ cpTqSp"
            for="darkMode"
          >
            Dark mode
          </label>
        </div>
        <button
          class="sc-fubCfw iHeQum"
        />
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
